### DISCUSSION TITLE
jsdelivr CDN fallback

### DISCUSSION THREAD

**Discussion by kira0269** [symfony/symfony#54801] [Polls]
Hi community,

Today is a hard day for all asset-mapper component lovers: the certificate of **cdn.jsveliver.net** has **expired**.

Because of that, the command `importmap:install` fails and deployment can't be done.

I found that there is a solution: calling another endpoint from the ones below:
- fastly.jsdelivr.net
- gcore.jsdelivr.net

Today, two files in the symfony sources contain the string "cdn.jsdelivr.net" (on from asset-mapper component and one from the mime component). They are **hardcoded** and the only way to override those is to apply a composer patch for example.

**Question 1:** which solutions could be the best to prevent this issue as best as possible?

Some ideas:

- Configure the other CDNs as fallbacks if the first one fails due to connection error
- Put into a configuration the list of available CDNs

**Question 2:** is it worth it to implement this logic? Are CDNs just not 100% reliable by definition and we just have to face those issues some day or another ?

**Comment by wotta** [comment]
To bundle all related information I will share the GitHub issue from jsdelivr and Cloudflare forum page.

- https://github.com/jsdelivr/jsdelivr/issues/18565
- https://community.cloudflare.com/t/jsdelivr-expired-certificate-may-2024/650543/4
- https://twitter.com/jsDelivr/status/1785944625158508789
